_posts/2025-10-23-mmaler-blogpost-2-quarkus-runtime-and-framework-for-cloud-native-java.adoc

```diff
@@ -0,0 +1,161 @@
+
+These features reflect Quarkus’s dual role: it behaves like a framework during development, offering structure, defaults, and conventions, and it acts as a runtime during execution, delivering performance, efficiency, and operational consistency.
+
+=== Performance that matters
+
+In practice, the two goals that matter most for production-grade frameworks at scale are fast startup and resilience against cascading failures.
+Rapid startup improves elasticity and recovery, while resilience prevents local faults from spreading.
+For a broader industry perspective, see the link:https://cacm.acm.org/practice/application-frameworks/[ACM article on application frameworks].
+
+Native image generation with GraalVM enables cold starts in milliseconds and a minimal memory footprint.
+As such, it allows cost-effective deployment in containers and serverless environments.
+
+Quarkus also includes structured health checks, metrics, and tracing, and thereby aligns with production standards from day one.
+
+
+image::7-success-stories.png[alt="Real-world success stories using Quarkus", width=100%]
+
+
+=== Security
+
+Quarkus uses a standards-first composable security model.
+You enable what you need and configure it for your environment:
+
+* **Transport:** Enable TLS for HTTPS, or terminate TLS at your ingress.
+* **Authentication:** Choose Basic, form-based, mTLS, or OpenID Connect (OIDC).
+* **Authorization:** Enforce RBAC on web endpoints with `@RolesAllowed`, `@DenyAll`, and `@PermitAll`.
+
+This lets you apply the right security controls for each deployment.
+
+=== Observability and control surfaces
+
+Common control surfaces, such as metrics, logging, tracing, and configuration, are essential for site reliability engineers (SREs) and platform teams.
+
+Quarkus exposes:
+
+* Unified logging with link:https://quarkus.io/guides/logging[`quarkus-logging`].
+* Structured metrics and tracing with link:https://quarkus.io/guides/telemetry-micrometer-to-opentelemetry[Micrometer and OpenTelemetry].
+* Unified configuration of all the application's aspects by using `application.properties` or environment variables.
+
+This standardization enables automation and scalable monitoring.
+
+=== Modular and production-ready
+
+Following a lean-core, modular-at-the-edge approach, Quarkus delivers:
+
+* A minimal core for fast startup.
+* Pluggable modules for authentication, tracing, messaging, and more.
+* Dev Services for automatic provisioning of databases, brokers, and containers in dev mode.
+
+Whether you are building a prototype or deploying to OpenShift, Quarkus adapts.
+This modularity spans both the framework-level APIs developers work with and the runtime behaviors that execute beneath them.
+
+Because Quarkus modularity is declarative and unified across extensions, it supports a platform-like developer experience without the rigidity of traditional frameworks.
+
+=== Building your stack with Quarkus
```

Comment:
I would actually write a full article about that part, as a "third" article in the series.